src/main/java/com/github/redouane59/twitter/TwitterClient.java

```diff
@@ -40,15 +40,7 @@
 import java.io.IOException;
 import java.net.URL;
 import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.Base64;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.NoSuchElementException;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```

Comment:
It feels a bit annoying to import using `*` wildcard.
You might force unnecessary qualified name usage on unused classes.

Reply:
Since IDE manages imports automatically I don't really care about this:) choice depends on code style. I use IntelliJ's default

Reply:
The problem you are describing doesn't exist. Specific import always overrides wildcard. e.g you can import com.youcompany.List and it will be used instead of java.util.List which iss included into the wildcard java.until.*

Reply:
What do you mean? I'll check this ASAP but it is even reported as a critical Sonar issue: https://rules.sonarsource.com/java/RSPEC-2208

Reply:
Critical for what? Can you explain potential problems (in the world where no-one manages imports manually:)). Why do we care about what Sonar (who they are?) considers critical? There are plain old Sun code conventions https://www.oracle.com/technetwork/java/codeconventions-150003.pdf that doesn't prohibit usages of wildcard import.. it's a part of the language:)
It's ok to follow the code style of the project, but , I haven't found any references in readme.md. 
Code style is somewhat important, but there are tons of really serious problems in the project that make it completely unusable in production. 
Let's make it work first.

Reply:
## Compilation

I have been conducting a pure Java test:

```java
import static io.vavr.control.Option.*;
import static io.vavr.Tuple.*;

import org.junit.jupiter.api.Test;

class Example {

  @Test
  void test() {

    String test = "hi";
    of(test);

  }
}
```

then ran a `mvn install`, and here's the output:

```log
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project onepay-psp-ingenico: Compilation failure
[ERROR] /home/hypr2771/Workspace/OnePay/onepay-psp-ingenico/src/test/java/com/decathlon/onepay/psp/ingenico/tools/impl/Example.java:[14,5] reference to of is ambiguous
[ERROR]   both method <T>of(T) in io.vavr.control.Option and method <T1>of(T1) in io.vavr.Tuple match
```

## Readability

When reading the following code:

```java
import io.vavr.collection.List;
import java.util.*;
import io.vavr.collection.*;
import org.junit.jupiter.api.Test;

class Example {

  @Test
  void test() {

    String test = "hi";
    List.of(test);
    new ArrayList<String>(1);

  }
}
```

it really does not seem intuitive to know which class is being used.

## Real case scenario

```files
- my-collections.1.0.jar
+- my.collection.List
```

```files
- my-collections.1.1.jar
+- my.collection.List implements Collection
+- my.collection.ArrayList implements List
```

When upgrading `my-collections` from 1.0 to 1.1, we will encounter a compilation issue which would just be avoided using fully qualified imports.

```java
import java.util.*;
import my.collection.*;
import org.junit.jupiter.api.Test;

class Example {

  @Test
  void test() {

    new ArrayList<String>(1);

  }
}
```

Even worse, if the `my-collections` implements the same interface, you expose yourself to a potential behavioral change if you import the wrong one after encountering the compilation issue.
Now I agree this should _never_ happen, as we supposedly always double check our imports, but it is still an issue which exists _only_ because we used wildcard imports first.

## Overall

As you said, imports are rarely managed manually, so why not leave IntelliJ never use wildcard?
This is easily set up in IntelliJ and have _absolutely no drawbacks_.

This is taking a bit too strong proportions, I agree, and this could indeed have been avoided writing a contribution and code style guide.
@redouane59 I suggest you work on such directives if you want your project to be contributed on by various developers.

Reply:
>I have been conducting a pure Java test
>then ran a mvn install, and here's the output:
So what? Like you are trying to show me if you put .of instead of * it'll compile or what? 

>it really does not seem intuitive to know which class is being used.
Whole code doesn't make any sense.. import is a lesser evil.

>As you said, imports are rarely managed manually, so why not leave IntelliJ never use wildcard?
Nothing, except there is defined code style in the project. There are tons of blockers in it and you are not approving major fixes because of cosmetics. 
At the moment I want to help with the project. If you are worried about the code style then here is a ticket for this https://github.com/redouane59/twittered/issues/78 take it and implement it as you wish. Add automated checks, configs for major ides and make everyone happy.
If you think that absence of code style is a blocker from merging let me know. Then  I'll just keep working on my fork.

Reply:
For this point I agree with Vincent, even if nothing about code style and good practices is present in this repo, I would prefer to let imports as they were before.

Reply:
@jkolobok just to clarify things up, I have _no_ rights on this repository, I am just checking up quality once in a while.
I agree with you, there are other issues, but it does not mean we accept to integrate lower quality issues.

Now, on this very specific issue, this can be fixed with correct style and optimizing imports. The sole point is that we want to make the code harmonious and style consistent. It seems we agree on this, so my guess would be that @redouane59 merges a PR with IntelliJ code style and some contribution guidelines first; before starting to merge contributors' PR.

Does it makes sense?